homeassistant/components/shelly/__init__.py

```diff
@@ -39,18 +45,36 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         entry.data.get(CONF_PASSWORD),
         temperature_unit,
     )
+
+    if COAP_CONTEXT not in hass.data[DOMAIN][DATA_GENERAL]:
+        coap_context = hass.data[DOMAIN][DATA_GENERAL][
+            COAP_CONTEXT
+        ] = await aiocoap.Context.create_client_context()
```

Comment:
Something is wrong here. I have 3 devices configured and `coap_context` is created 3 times during HA startup.

Reply:
All entries are set up in parallel. You need to use a lock. But yeah, for now maybe just easier to do this in `async_setup` and call it a day for now 👍